### PR TITLE
th-26: Add toast notifications service

### LINKED ISSUE #26: Toast notifications
Use React-Toastify.

Create the Notifications service which will accumulate all React-Toastify abilities in a single place.

### PR DESCRIPTION
**Pull request by denred**
- Installed the "react-toastify" package to manage notifications.

- Added a new notification service for convenient display of notifications.

- Exported an instance of the notification service into the relevant part of the application.

- Set up default options to configure the standard behavior of notifications.

### CODE REVIEW

File: frontend/src/index.tsx [hunk 1]
```diff
@@ -38,5 +40,6 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
         ]}
       />
     </StoreProvider>
+    <ToastContainer />
```
Comment (kalinkaaaa14) [inline frontend/src/index.tsx]: create a wrapper for toast container and place this wrapper somewhere in components, it should not be placed in index file

File: frontend/src/libs/packages/notification/libs/options/notification.options.default.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { type ToastOptions, toast } from 'react-toastify';
+
+import { NotificationTheme } from '../enums/notification.theme.enum.js';
+
+const defaultNotificationOptions: ToastOptions = {
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/libs/options/notification.options.default.ts]:
```suggestion
const DEFAULT_NOTIFICATION_OPTIONS: ToastOptions = {
```

File: frontend/src/libs/packages/notification/notification.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { defaultNotificationOptions } from './libs/options/notification.options.default.js';
+import { NotificationService } from './notification-api.js';
+
+const notification = NotificationService.getInstance(
+  defaultNotificationOptions,
+);
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/notification.ts]: I think we don't need get instance method, use public constructor instead
const notification = new NotificationService(options);

File: frontend/src/libs/packages/notification/notification-api.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    toast.info(message, { ...this.#options, ...options });
+  }
+
+  public static getInstance(options: ToastOptions): NotificationService {
+    if (!this.#instance) {
+      this.#instance = new NotificationService(options);
+    }
+
+    return this.#instance;
+  }
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/notification-api.ts]: think we can get rid of it

File: frontend/src/libs/packages/notification/libs/options/notification.options.default.ts (hunk 2 above)
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/libs/options/notification.options.default.ts]: also create separate type for toast options, describe inside only those which we are planning to use. Now a lot of default options I suppose we will not use

File: frontend/src/libs/packages/notification/libs/enums/notification.theme.enum.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { type Theme } from 'react-toastify';
+
+const NotificationTheme = {
+  LIGHT: 'light' as Theme,
+  DARK: 'dark' as Theme,
+  COLORED: 'colored' as Theme,
+};
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/libs/enums/notification.theme.enum.ts]:
```suggestion
const NotificationTheme: TypeHere = {
  LIGHT: 'light',
  DARK: 'dark',
  COLORED: 'colored',
} as const;
```

File: frontend/src/libs/packages/notification/notification-api.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { type ToastOptions, toast } from 'react-toastify';
+
+class NotificationService {
+  #options: ToastOptions;
+
+  static #instance: NotificationService | undefined;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/notification-api.ts]: can get rid of

**Review by kalinkaaaa14 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/notification/notification.tsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/notification/notification.tsx]: do we need this import here?

File: frontend/src/libs/components/notification/notification.tsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+
+import { type ToastContainerProps } from 'react-toastify';
+import { ToastContainer } from 'react-toastify';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/notification/notification.tsx]: should be in 1 import

File: frontend/src/libs/packages/notification/libs/enums/notification.theme.enum.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+const theme = {
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/libs/enums/notification.theme.enum.ts]:
```suggestion
const NotificationTheme= {
```
also file naming - notification-theme.enum

**Review by kalinkaaaa14 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/packages/notification/libs/options/notification.options.default.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { type Theme, type ToastPosition, toast } from 'react-toastify';
+
+import { theme } from '../enums/notification.theme.enum.js';
+
+type IToastOptions = {
+  position?: ToastPosition;
+  theme?: Theme;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/libs/options/notification.options.default.ts]: you used Theme from the library in type, but in default options - created theme enum. I guess it should be custom enum everywhere

File: frontend/src/libs/packages/notification/libs/options/notification.options.default.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  autoClose?: number;
+};
+
+const DEFAULT_NOTIFICATION_OPTIONS: IToastOptions = {
+  position: toast.POSITION.TOP_RIGHT,
+  theme: theme.LIGHT,
+  autoClose: 4000,
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/notification/libs/options/notification.options.default.ts:19]: I think we need to add also transition

**Review by kalinkaaaa14 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/notification/notification.tsx (hunk 7 above)
Comment (denred) [reply frontend/src/libs/components/notification/notification.tsx]: should styles be imported within the component or at the time the component is used, and which approach is better?

File: frontend/src/libs/packages/notification/libs/options/notification.options.default.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import {
+  type ToastPosition,
+  type ToastTransition,
+  toast,
+} from 'react-toastify';
+
+import { NotificationTheme, NotificationTransition } from '../enums/enums.js';
+
+type IToastOptions = {
```
Comment (PavloSukhinin) [inline frontend/src/libs/packages/notification/libs/options/notification.options.default.ts:9]: Let's name types without `I` prefix

File: frontend/src/libs/packages/notification/libs/options/notification.options.default.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  position?: ToastPosition;
+  theme?: (typeof NotificationTheme)[keyof typeof NotificationTheme];
```
Comment (PavloSukhinin) [inline frontend/src/libs/packages/notification/libs/options/notification.options.default.ts:11]:
```suggestion
  theme?: ValueOf<typeof NotificationTheme>;
```

**Review by PavloSukhinin — CHANGES_REQUESTED** [review]
Don't forget to pass instance of NotificationService as extra arguments for Thunks
`frontend/src/libs/packages/store/store.package.ts`

**Pull request closed without merging**